test/AutoDiff/Serialization/derivative_attr_cross_module.swift

```diff
@@ -0,0 +1,11 @@
+// REQUIRES: asserts
+
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module -module-name Multi -o %t/multi-file.swiftmodule -primary-file %s %S/Inputs/derivative_attr_cross_module.swift
+// RUN: %target-swift-frontend -emit-module -module-name Multi -o %t/multi-file-2.swiftmodule %s -primary-file %S/Inputs/derivative_attr_cross_module.swift
+
+// Check that merge-modules does not crash regardless of the order of inputs.
+// RUN: %target-swift-frontend -emit-module -module-name Multi %t/multi-file.swiftmodule %t/multi-file-2.swiftmodule -o %t -print-stats 2>&1
+// RUN: %target-swift-frontend -emit-module -module-name Multi %t/multi-file-2.swiftmodule %t/multi-file.swiftmodule -o %t -print-stats 2>&1
+
+func id(_ x: Float) -> Float { x }
```

Comment:
Unless I'm mistaken, this shouldn't really crash. You should try to serialize something that cross-references itself.

Reply:
Yes, this test didn't crash before this patch.

> You should try to serialize something that cross-references itself.

I can't think of a good example, do you have ideas for a test? I unsuccessfully tried to adapt [`test/Serialization/multi-file-type-eraser.swift`](https://github.com/apple/swift/blob/master/test/Serialization/multi-file-type-eraser.swift).

Reply:
What if you declare one function in each file that are each other's derivatives?